src/mpl2/src/rtl_mp.cpp

```diff
@@ -156,7 +156,9 @@ bool rtl_macro_placer(const char* config_file,
   int max_num_step = 4000;
   int perturb_per_step = 300;
 
-  int snap_layer = 4;
+  // Platform (and design macro specific)
+  int snap_layer_x = 4;
+  int snap_layer_y = 5;
```

Comment:
You should at least check the preferred routing direction when you pick these.

Reply:
Will handle this in a sep PR Want to merge the auto cluster changes and the macro dimension issue fix